tests/schema/test_enum.py

```diff
@@ -334,3 +334,38 @@ def best_flavour(self) -> IceCreamFlavour:
 
     assert not result.errors
     assert result.data["bestFlavour"] == "STRAWBERRY"
+
+
+def test_enum_deprecated_value():
+    @strawberry.enum
+    class IceCreamFlavour(Enum):
+        VANILLA = "vanilla"
+        STRAWBERRY = strawberry.enum_value(
+            "strawberry", deprecation_reason="We ran out"
+        )
+        CHOCOLATE = strawberry.enum_value("chocolate")
+
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def best_flavour(self) -> IceCreamFlavour:
+            return IceCreamFlavour.STRAWBERRY
+
+    schema = strawberry.Schema(query=Query)
+
+    query = """
+    {
+        __type(name:"IceCreamFlavour") {
+            enumValues {
+                isDeprecated
+                deprecationReason
+            }
+        }
+    }
+    """
+
+    result = schema.execute_sync(query)
+
+    assert not result.errors
+    # deprecated value dissapears
+    assert len(result.data["__type"]["enumValues"]) == 2
```

Comment:
@patrick91 @BryceBeagle 

Any idea why deprecated value disappears? Am I missing something obvious?

Reply:
I've updated the query to include deprecated values :)

Reply:
🤦‍♂️ thanks totally forgot that you need that in introspection, thanks a lot :)